### PR TITLE
Quantity box

### PR DESCRIPTION
**Pull request by spireken**

### CODE REVIEW

**Review by shaobinspire — APPROVED** [review]
Looks good.

File: Applications/Spire/Include/Spire/Ui/QuantityBox.hpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    ScalarValueModel<boost::optional<Nexus::Quantity>>;
+
+  /** A LocalScalarValueModel over Quantity values. */
+  using LocalQuantityModel = LocalScalarValueModel<Nexus::Quantity>;
+
+  /** A LocalScalarValueModel over optional Quantity values. */
+  using LocalOptionalQuantityModel =
+    LocalScalarValueModel<boost::optional<Nexus::Quantity>>;
+
+  class QuantityBox : public DecimalBoxAdaptor<Nexus::Quantity> {
```
Comment (kranar) [inline Applications/Spire/Include/Spire/Ui/QuantityBox.hpp:23]: Document.

File: Applications/Spire/Include/Spire/Ui/QuantityBox.hpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    public:
+
+      /**
+       * Constructs a QuantityBox with a LocalValueModel.
```
Comment (kranar) [inline Applications/Spire/Include/Spire/Ui/QuantityBox.hpp:27]: LocalValueModel -> LocalOptionalQuantityModel

File: Applications/Spire/Include/Spire/Ui/QuantityBox.hpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+       * @param modifiers The keyboard modifier increments.
+       * @param parent The parent widget.
+       */
+      explicit QuantityBox(QHash<Qt::KeyboardModifier,
```
Comment (kranar) [inline Applications/Spire/Include/Spire/Ui/QuantityBox.hpp:31]: Break lines up at the higher nesting level.

File: Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp [hunk 4]
```diff
@@ -42,6 +43,29 @@ using namespace Spire;
 using namespace Spire::Styles;
 
 namespace {
+
```
Comment (kranar) [inline Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp:46]: Remove vertical whitespace.

File: Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp [hunk 5]
```diff
@@ -42,6 +43,29 @@ using namespace Spire;
...
+  template<typename T>
```
Comment (kranar) [inline Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp:47]: Daaamn this is nieeece.

File: Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp [hunk 6]
```diff
@@ -42,6 +43,29 @@ using namespace Spire;
...
+
+    Type m_current;
+    Type m_minimum;
+    Type m_maximum;
+    Type m_default_increment;
+    Type m_alt_increment;
+    Type m_ctrl_increment;
+    Type m_shift_increment;
+
+    DecimalBoxProfileProperties(Type default_increment)
```
Comment (kranar) [inline Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp:59]: ```explicit```

File: Applications/Spire/Source/UiViewer/StandardUiProperties.cpp [hunk 7]
```diff
@@ -156,7 +156,7 @@ std::shared_ptr<TypedUiProperty<Money>> Spire::make_standard_property<Money>(
   return std::make_shared<StandardUiProperty<Money>>(std::move(name), value,
     [] (QWidget* parent, StandardUiProperty<Money>& property) {
       auto setter = new QDoubleSpinBox(parent);
-      setter->setMinimum(std::numeric_limits<double>::min());
+      setter->setMinimum(std::numeric_limits<double>::lowest());
```
Comment (kranar) [inline Applications/Spire/Source/UiViewer/StandardUiProperties.cpp:159]: Niiccce.

**Review by kranar — CHANGES_REQUESTED** [review]
Sweeet

File: Applications/Spire/Source/UiViewer/StandardUiProperties.cpp (hunk 7 above)
Comment (spireken) [reply Applications/Spire/Source/UiViewer/StandardUiProperties.cpp:159]: I have to give credit to @shaobinspire for pointing this out.

File: Applications/Spire/Include/Spire/Ui/QuantityBox.hpp (hunk 3 above)
Comment (spireken) [reply Applications/Spire/Include/Spire/Ui/QuantityBox.hpp:31]: Can you explain what you mean? I can't find an issue with this formatting.

Comment (kranar) [reply Applications/Spire/Include/Spire/Ui/QuantityBox.hpp:31]: explicit QuantityBox(
      QHash<Qt::KeyboardModifier, Nexus::Quantity> modifiers, QWidget* parent = nullptr);

**Review by kranar — APPROVED** [review]

**Comment by spireken** [comment]
I have to fix a compilation error regarding the now explicit constructor for the struct I added.

**Pull request merged**